Makefile

```diff
@@ -43,5 +43,18 @@ firsttime:
 update:
 	rustup update
 
+# Option used for formatting. If set, the command will only verify if
+# formatting is correct (without actually changing the formatting).
+# Returns 0 only if all files are properly formatted.
+# Usage:
+# 	$ make format -k check=true
+check ?=
+
+CRATES_TO_FORMAT := $(shell find . -name "Cargo.toml")
```

Comment:
I'd really like to avoid finds like this if possible. we're starting down the same messy path that the coreboot makefiles followed. The struct of the oreboot directory tree is highly constrained (intentionally); is there some way you can just glob, e.g. src/*/*/*/Cargo.toml

I've been struggling with coreboot make complexity a lot the last few months. I know this does not seem like much but in a few years time, if we are not careful, we're going to have the same makefile mess coreboot has today.

consider this: the goal of oreboot is to allow someone who is not a fulltime firmware person to build and use firmware. it's essential that we keep things simple. Is there some way to make this a bit less so.

Reply:
Makefile does not support recursive "wildcard", but if we assume that max depth of any Cargo crate would be e.g. 4, then we can just call wildcard 4 times. It is a little longer (and depends on max depth), but is probably much easier to understand (does not depend on understanding custom commands like find).

On similar note, if you do not like the "check" parameter, and you would prefer to just use additional target (e.g. `check_format` or `check.format` or `ci.format`), please let me know and I can try to update it as well (but this might not be as trivial as it seems, and in the end could complicate makefile more than the variable). But if you want I can at least try it out and we can choose which solution is preferred.